vowpalwabbit/parse_args.cc

```diff
@@ -1272,82 +1274,135 @@ VW::LEARNER::base_learner* setup_base(options_i& options, vw& all)
     return base;
 }
 
+void create_reduction_template(options_i& options, vw& all)
+{
+  // TODO: find a better way.
+  // list of all reductions that require a multilearner as its base reduction
+  static const std::unordered_set< VW::LEARNER::base_learner* (*)(VW::config::options_i&, vw&)> multi = {
+    explore_eval_setup
+    , cbifyldf_setup
+    , cbify_setup
+    , warm_cb_setup
+    , VW::slates::slates_setup
+    , CCB::ccb_explore_adf_setup
+    , VW::shared_feature_merger::shared_feature_merger_setup
+    , cb_sample_setup
+    , cb_dro_setup
+    , VW::cb_explore_adf::greedy::setup
+    , VW::cb_explore_adf::softmax::setup
+    , VW::cb_explore_adf::rnd::setup
+    , VW::cb_explore_adf::regcb::setup
+    , VW::cb_explore_adf::first::setup
+    , VW::cb_explore_adf::cover::setup
+    , VW::cb_explore_adf::bag::setup
+    , cb_adf_setup
+    , cb_adf_setup
+  };
+  std::deque<VW::LEARNER::base_learner* (*)(VW::config::options_i&, vw&)> tmp_reduction_stack;
+  std::swap(tmp_reduction_stack, all.reduction_stack);
+
+  auto *cast_options = dynamic_cast<options_boost_po*>(&options);
+  auto options_str = cast_options->m_command_line;
+  options_i* tmp_options = new options_boost_po(options_str);
+
+  for (auto reduction_it = tmp_reduction_stack.rbegin(); reduction_it != tmp_reduction_stack.rend() - 11; ++reduction_it) {
```

Comment:
11? could we give 11 a name?

Reply:
The `-11` is only there to avoid calling the base reduction setup functions here. And the only reason we need to do that is because some of them do things that will break if the setup is called multiple times. Final implementation should not have that at all (I'll add a comment)